demos/defender/defender_demo_json/report_builder.c

```diff
@@ -479,6 +670,41 @@ ReportBuilderStatus_t GenerateJsonReport( char * pBuffer,
         }
     }
 
+    /* Write custom metrics. */
```

Comment:
Combining the 5 macros into one could reduce all this code into:
```c
#define JSON_REPORT_FORMAT_PART5    \
    "\"custom_metrics\":{"          \
    "\"cpu-usage\":["               \
    "{"                             \
    "\"number_list\":["             \
    "%lu,"                          \
    "%lu"                           \
    "]"                             \
    "}"                             \
    "],"                            \
    "\"memory-info\":["             \
    "{"                             \
    "\"number_list\":["             \
    "%u,"                           \
    "%u"                            \
    "]}"                            \
    "]"                             \
    "}"                             \
```

```c
    /* Write custom metrics. */
    if( status == ReportBuilderSuccess )
    {
        charactersWritten = snprintf( pCurrentWritePos,
                                      remainingBufferLength,
                                      JSON_REPORT_FORMAT_PART5,
                                      pMetrics->pCustomMetrics->cpuUsageStats.upTime,
                                      pMetrics->pCustomMetrics->cpuUsageStats.idleTime,
                                      pMetrics->pCustomMetrics->memoryStats.totalMemory,
                                      pMetrics->pCustomMetrics->memoryStats.availableMemory );

         if( !SNPRINTF_SUCCESS( charactersWritten, remainingBufferLength ) )
        {
            LogError( ( "Failed to write part 5." ) );
            status = ReportBuilderBufferTooSmall;
        }
        else
        {
            remainingBufferLength -= charactersWritten;
            pCurrentWritePos += charactersWritten;
        }
    }
```

Reply:
Done